packages/validator/src/validate.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs';
+import * as logger from './utils/logger';
+import { Result } from './types';
+import { validateSpecs } from './validator';
+
+/**
+ * Validates specification from provided file according to template file
+ * @param specsPath - specification to validate, root must be an object (not an array)
+ * @param templatePath - validator specification structure
+ * @returns array of error and warning messages
+ */
+export function validate(specsPath: string | undefined, templatePath: string | undefined): Result {
+  if (!specsPath || !templatePath) {
+    return { valid: false, messages: [logger.error('Specification and template file must be provided')] };
+  }
+
+  let template, specs;
+
+  try {
+    template = fs.readFileSync(templatePath);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${templatePath}`)] };
+  }
+
+  try {
+    specs = fs.readFileSync(specsPath);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${specsPath}`)] };
+  }
+
+  try {
+    const nonRedundant = {};
+    const parsedTemplate = JSON.parse(template);
+    const parsedSpecs = JSON.parse(specs);
+
+    return validateSpecs(parsedSpecs, parsedTemplate, '', nonRedundant, {
+      specs: parsedSpecs,
+      nonRedundantParams: nonRedundant,
+    });
+  } catch (e) {
+    return { valid: false, messages: [{ level: 'error', message: `${e.name}: ${e.message}` }] };
+  }
+}
+
+if (process.env.npm_config_specs || process.env.npm_config_template) {
```

Comment:
Using env variables isn't mentioned in the README. Worth adding it there I think.

Also, I think let's drop the `npm_` prefix as it implies it's an NPM setting and uppercase the name for consistency.

Reply:
Its not supposed to be used as environment variables, its how npm passes arguments when running script with arguments in `--variable="value"` format, the value of `variable` can be accessed as `process.env.npm_config_variable`. Thats the reason no environment variables are mentioned in readme and prefix must stay.

Reply:
Wasn't aware of that behaviour. Good to know, thanks